GenderPayGap.WebUI/Search/SearchRepository.cs

```diff
@@ -120,18 +124,26 @@ private static List<SearchCachedOrganisation> LoadAllOrganisations(IDataReposito
                                 EmployerReference = o.EmployerReference?.Trim(),
                                 OrganisationNames =
                                     o.OrganisationNames.OrderByDescending(n => n.Created)
-                                        .Select(on => new SearchReadyValue(@on.Name))
+                                        .Select(on => new SearchReadyValue(on.Name))
                                         .ToList(),
                                 MinEmployees = o.GetLatestReturn()?.MinEmployees ?? 0,
                                 Status = o.Status,
-                                OrganisationSizes = o.Returns.Where(r => r.Status == ReturnStatuses.Submitted).Select(r => r.OrganisationSize).Distinct().ToList(),
+                                ReportingYearToOrganisationSizesMap = submittedReports
+                                    .GroupBy(r => r.AccountingDate.Year)
+                                    .ToDictionary(
+                                        g => g.Key,
+                                        g => g.ToList().Select(r => r.OrganisationSize).Distinct().ToList()),
                                 SicSectionIds =
                                     o.OrganisationSicCodes.Select(osc => Convert.ToChar(osc.SicCode.SicSection.SicSectionId)).ToList(),
-                                ReportingYears = o.Returns.Where(r => r.Status == ReturnStatuses.Submitted).Select(r => r.AccountingDate.Year).ToList(),
-                                DateOfLatestReport =
-                                    o.GetLatestReturn() != null ? o.GetLatestReturn().StatusDate.Date : new DateTime(1999, 1, 1),
-                                ReportedWithCompanyLinkToGpgInfo = o.Returns.Where(r => r.Status == ReturnStatuses.Submitted).Any(r => r.CompanyLinkToGPGInfo != null),
-                                ReportedLate = o.Returns.Where(r => r.Status == ReturnStatuses.Submitted).Any(r => r.IsLateSubmission),
+                                ReportingYears = submittedReports.Select(r => r.AccountingDate.Year).ToList(),
+                                ReportingYearToDateOfLatestReportMap = ReportingYearsHelper.GetReportingYears()
+                                    .ToDictionary(
+                                        y => y,
+                                        y => o.GetReturn(y) != null ? o.GetReturn(y).StatusDate.Date : new DateTime(1999, 1, 1)),

```

Comment:
would this make more sense to be Global.FirstReportingYear?

Reply:
not really, what if the employer has no reports for the first reporting year?